<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>ViCTree</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-light.css">
    <meta name="viewport" content="width=device-width">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>ViCTree</h1>
        <p>A Virus Taxonomy Classification Framework</p>

        <p class="view"><a href="https://github.com/josephhughes/ViCTree">View the Project on GitHub <small>josephhughes/ViCTree</small></a></p>


        <ul>
          <li><a href="https://github.com/josephhughes/ViCTree/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/josephhughes/ViCTree/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/josephhughes/ViCTree">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
    

<p><a href="http://josephhughes.github.io/ViCTree/">ViCTree Home </a></p>
<h3>
<a id="welcome-to-github-pages" class="anchor" href="#welcome-to-github-pages" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Densovirinae example</h3>
<p> ViCTree pipeline is currently set up for Densovirinae sub-family. Trees and alignments are updated when new sequences are submitted to the Genbank and if the newly added sequences form a new cluster that is not previously identified. </p>
<p> <i> Densovirinae </i> subfamily analysis is based on the Non structural protein 1 sequences. <a href = "https://github.com/josephhughes/ViCTree/blob/master/txid40120_seeds.fa"> 21 seed sequences </a> were used to carry out initial as well all subsequent analysis for the subfamily level taxonomic classification of the viruses. Pre-optimised blast parameter for a hit length 100 <code>(-l 100)</code> with the coverage 50 <code>(-c 50)</code> was specified.
Clustering threshold of 0.9 was applied to cluster sequences that were 90% or more identical. There is an optional parameter <code> -u </code>to specify a file with a list of protein accession numbers that are accepted by ICTV as representative species for the family or the subfamily. This option allows the users to keep the static branches expanding phylogeny consistent.</p> 
<p>Multiple sequence alignment and pairwise distance are calculated for the sequences included in the final set after the clustering step. Sequence metadata information is collected for this set of sequences. This includes the information about the genome accession, scientific name,	taxonomy ID, taxonomy 
lineage, genus and NCBI	URL. This information enables users to customise labels for the tips of the tree in the visualisation module. It also provide a link-out to NCBI genome sequence page for each sequence represented in the tree. A tree with rapid bootstrap analysis and best-scoring maximum likelihood with PROTGAMMAJTT (default) model is generated using RAxML. </p>
<p>The tree with the pairwise distance matrix and the metadata tables are then automatically submitted to the Phylotree module of the pipeline for the visualisation.</p>

<table>
<thead>
<tr>
<th>Filename</th>
<th align="center">Contents</th>
</tr>
</thead>
<tbody>
<tr>
<td> <a href ="https://github.com/josephhughes/ViCTree/blob/master/txid40120_seeds.fa"> Seed set </a></td>
<td align="center">All protein sequences used as seed set for <i> Densovirinae </i> example.</td>
</tr>
<tr>
<td> <a href ="https://github.com/josephhughes/ViCTree/blob/master/txid40120/txid40120_final_set.fa" > Final set </a></td>
<td align="center">The final set of sequences selected after BLAST and Clustering step. This set is used for Multiple Sequence Alignment.</td>
</tr>
<tr>
<td> <a href ="https://github.com/josephhughes/ViCTree/blob/master/txid40120/txid40120_final_set_clustalo_aln.fa"> Alignments</a></td>
<td align="center">Protein sequence alignments for the final set of sequences.</td>
<tr>
<td> <a href ="https://github.com/josephhughes/ViCTree/blob/master/txid40120/txid40120.csv"> Pairwise distance matrix </a></td>
<td align="center"> The pairwise distance matrix for the final set of sequences.</td>
</tr>
<tr>
<td> <a href ="https://github.com/josephhughes/ViCTree/blob/master/txid40120/txid40120.nhx"> Phylogenetic tree </a></td>
<td align="center">Phylogenetic tree generated by the ViCTree pipeline in the newick format. </td>
</tbody>
</table>

<p> Following figure shows a snapshot of the <i> Densovirinae </i> tree generated by the ViCTree analysis, visualised in the Phylotree module of the framework. </p>
<figure>
	<img align="left" src="images/Densovirinae_ViCTree.png"/>
	<figcaption> ViCTreeâ€™s phylotree visualisation on the  <a href="www.bioinformatics.cvr.ac.uk/victree">CVR Bioinformatics</a> . It shows the Densovirinae subfamily level taxonomic classification tree generated by ViCTree pipeline. Top bar from left to right labelled: a. Browse between examples. b. Choose a number of options for the tree tip labels. c. Download the tree image in SVG and PNG formats. d. Expand the gaps between tree branches. e. Shrink the gap between the tree branches. f. Switch between the ultrametric tree and the view with the branch lengths. g. A textbox and a scrollbar to select the pairwise distances between sequences.
	</figcaption>
</figure>
<p>The ViCTree framework is developed by : </p>
<p>Joseph Hughes (<a href="https://github.com/josephhughes" class="user-mention">@josephhughes</a>), Sejal Modha (<a href="https://github.com/sejmodha" class="user-mention">@sejmodha</a>) and Anil Thanki (<a href="https://github.com/anilthanki" class="user-mention">@anilthanki</a>) </p>


  </body>
</html>
