<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>ViCTree</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-light.css">
    <meta name="viewport" content="width=device-width">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>ViCTree</h1>
        <p>A Virus Taxonomy Classification Framework</p>

        <p class="view"><a href="https://github.com/josephhughes/ViCTree">View the Project on GitHub <small>josephhughes/ViCTree</small></a></p>


        <ul>
          <li><a href="https://github.com/josephhughes/ViCTree/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/josephhughes/ViCTree/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/josephhughes/ViCTree">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
    

<p><a href="http://josephhughes.github.io/ViCTree/">ViCTree Home </a></p>
<h3>
<a id="welcome-to-github-pages" class="anchor" href="#welcome-to-github-pages" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Installation</h3>
<p> ViCTree can be installed on any UNIX/LINUX machine. This pipeline is tested and works on Ubuntu machines with multiple cores. </p>

<h3>
<a id="welcome-to-github-pages" class="anchor" href="#welcome-to-github-pages" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Prerequisites</h3>
<p> The following programs must be available in the $PATH environment. </p>
<p><ul>
<li><a href="http://www.bioperl.org/wiki/Main_Page">BioPerl </a> and<a href="http://search.cpan.org/~cjfields/Bio-EUtilities/"> Bio::DB::EUtilities module</a></li>
<li><a href="http://www.clustal.org/omega/">Clustal-Omega</a></li>	
<li><a href="https://www.gnu.org/software/bash/">GNU BASH environment</a></li>	
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/legacy/2.2.26/">Legacy BLAST (blastall,formatdb)</a></li>	
<li><a href="http://sco.h-its.org/exelixis/web/software/raxml/hands_on.html">RAxML-PThreads</a></li>	
</ul>
</p>

<p> Download ViCTree package or clone ViCTree from this page and make sure all scripts are available in $PATH. </p>
<h3>
<a id="running ViCTree" class="anchor" href="#authors-and-contributors" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Running ViCTree</h3>
<p> To run the ViCTree pipeline you will need to specify 2 prior parameters </p>
<p><ul>
<li>The NCBI taxID of virus family or sub-family (e.g. 40120)</li>
<li>The seed set of sequences in a .fasta formatted file - This file should contain fasta protein sequences representing the known diversity of the virus family or sub-family of interest</li>
</ul>
</p>
<p>The following command will launch the ViCTree analysis pipeline for <I>Densovirinae</I> sub-family </p>
<pre><code>ICTV_pipeline.sh -t 40120 -s txid40120_seeds.fa -l 100 -c 50 -p 10 </code></pre>

<h3>
<a id="Output" class="anchor" href="#authors-and-contributors" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Output</h3>
<p> When the pipeline is run for the first time a folder with the taxID name is created that saves all the output files generated by the pipeline. </p>
<p> The main output files generated by the pipeline include:</p>
<!--<ul>
<li>taxID.fa - All protein sequences downloaded from NCBI for specified taxID</li>
<li>taxID_final_set.fa - Final set of sequences used for Multiple Sequence Alignment </li>
<li>taxID_seq_info - GIs grouped together representing the exact same sequence </li>
<li>taxID_tree.nhx - Final tree generated and rerooted by RAxML </li>
<li>taxID_clustalo_dist_mat.csv - Pairwise distance matrix file </li>
<li>taxID_metadata - Metadata file for each sequence from taxID_final_set.fa. Columns represent: Protein_GI,Nucleotide_GI,Genome_Accession,Species_Name,Description,Lineage,Genus,URL </li>
</ul> -->
<table>
<thead>
<tr>
<th>Filename</th>
<th align="center">Contents</th>
</tr>
</thead>
<tbody>
<tr>
<td>taxID.fa</td>
<td align="center">All protein sequences downloaded from NCBI for the specified taxID</td>
</tr>
<tr>
<td>taxID_final_set.fa</td>
<td align="center">The final set of sequences used for Multiple Sequence Alignment</td>
</tr>
<tr>
<td>taxID_seq_info</td>
<td align="center">GIs grouped together representing the same sequence</td>
</tr>
<tr>
<td>taxID_tree.nhx</td>
<td align="center">The final tree generated and rerooted by RAxML</td>
</tr>
<tr>
<td>taxID_clustalo_dist_mat.csv</td>
<td align="center">The pairwise distance matrix file </td>
</tr>
<tr>
<td>taxID_metadatav</td>
<td align="center">The metadata file for each sequence from taxID_final_set.fa that exists in NCBI </td>
</tr>
</tbody>
</table>

<p>The ViCTree framework is developed by : </p>
<p>Joseph Hughes (<a href="https://github.com/josephhughes" class="user-mention">@josephhughes</a>), Sejal Modha (<a href="https://github.com/sejmodha" class="user-mention">@sejmodha</a>) and Anil Thanki (<a href="https://github.com/anilthanki" class="user-mention">@anilthanki</a>) </p>


<!-- 
<h3>
<a id="support-or-contact" class="anchor" href="#support-or-contact" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Support or Contact</h3>

<p>Having trouble with Pages? Check out our <a href="https://help.github.com/pages">documentation</a> or <a href="https://github.com/contact">contact support</a> and weâ€™ll help you sort it out.</p>
      </section>
      <footer>
        <p>This project is maintained by <a href="https://github.com/josephhughes">josephhughes</a> and <a href="https://github.com/sejmodha">sejmodha</a> </p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>
              <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-75119550-1");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>
 -->

  </body>
</html>
