<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>ViCTree</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-light.css">
    <meta name="viewport" content="width=device-width">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>ViCTree</h1>
        <p>A Virus Taxonomy Classification Framework</p>

        <p class="view"><a href="https://github.com/josephhughes/ViCTree">View the Project on GitHub <small>josephhughes/ViCTree</small></a></p>


        <ul>
          <li><a href="https://github.com/josephhughes/ViCTree/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/josephhughes/ViCTree/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/josephhughes/ViCTree">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
    

<p><a href="http://josephhughes.github.io/ViCTree/">ViCTree Home </a></p>
<h3>
<a id="welcome-to-github-pages" class="anchor" href="#welcome-to-github-pages" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Installation</h3>
<p> ViCTree can be installed on any UNIX/LINUX machine. This pipeline is tested and works on Ubuntu machines with multiple cores. </p>
<p><a href="https://help.github.com/articles/fork-a-repo/"> Fork </a>the repository to add to your GitHub account. This is required for the visualisation as the tree data is fetched from the user's GitHub account. For more information see: <i> <b>Setting up GitHub repository for visualisation </i> </b> on this page.</p>
<p> ViCTree can be downloaded from GitHub using <a href="https://help.github.com/articles/cloning-a-repository/">clone </a> utility or can be downloaded as a zip file. </p>
<p> A sample command to clone ViCTree </p>
<pre> <code> git clone --recursive https://github.com/josephhughes/ViCTree.git </code> </pre> 
<b><i>Note:</i></b> Please replace the URL with your ViCTree repository URL for data upload on GitHub to work at the end of the pipeline.
<p></p>

<h3>
<a id="welcome-to-github-pages" class="anchor" href="#welcome-to-github-pages" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Prerequisites</h3>
<p> The following programs must be available in the $PATH environment. </p>
<p><ul>
<li><a href="http://bioperl.org/INSTALL.html">BioPerl </a> and<a href="ftp://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/"> NCBI Unix EUtilities</a></li>
<!-- <li><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/legacy/2.2.26/">Legacy BLAST (blastall, formatdb)</a></li> -->
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/">BLAST+ (blastp, makeblastdb)</a></li>
<li><a href="https://github.com/weizhongli/cdhit">cd-hit </a></li>
<li><a href="http://www.clustal.org/omega/">Clustal-Omega</a></li>	
<li><a href="https://git-scm.com/book/en/v2/Getting-Started-Installing-Git">Git</a></li>
<li><a href="https://www.gnu.org/software/bash/">GNU BASH environment</a></li>	
<li><a href="http://sco.h-its.org/exelixis/web/software/raxml/hands_on.html">RAxML-PThreads</a></li>	
</ul>
</p>

<p> Download ViCTree package or clone ViCTree from this page and make sure all scripts are available in $PATH. </p>
<h3>
<a id="running ViCTree" class="anchor" href="#authors-and-contributors" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Running ViCTree</h3>
<pre> <code> Usage: ViCTree [OPTIONS]
		
		-t Taxa ID - INT(Required) 
		-s Seedset in fasta format (Required) 
		-l Hit Length for BLAST - INT(Required) 
		-c Coverage for BLAST -INT(Required) 
		-h This helpful message
		-m Specify model for RAxML (Default is PTRGAMMJTT)
		-i Identity for clustering sequences using cdhit 
		-n Output name of the virus family or sub-family 
		-p Number of threads
		-u User-defined list of accession numbers to be set as cluster representatives
</code> </pre>

<p> To run the ViCTree pipeline you will need to identify two mandatory parameters </p>
<p><ul>
<li>The NCBI taxonomy ID of virus family or sub-family (e.g. 40120)</li>
<li>The seed set of sequences in a .fasta formatted file - This file should contain fasta protein sequences representing the known diversity of the virus family or sub-family of interest. 
<p>It is recommended to use protein accession number as header in order to retrieve metadata from NCBI as part of the pipeline. </p></li>
</ul>
</p>
<p>The following command will launch the ViCTree analysis pipeline for <I>Densovirinae</I> sub-family </p>
<pre><code>ViCTree.sh -t 40120 -s txid40120_seeds.fa -l 100 -c 50 -p 10  -i 0.9 -n Densovirinae</code></pre>

<h3> Setting up GitHub repository for visualisation
</h3>
Update <code>user_id, repo_name </code> and <code>branch </code> parameters to your forked repository in the following lines in the file <a href="https://github.com/anilthanki/phylotree/blob/master/index.html">index.html </a> in phylotree sud-directory.
<pre><code>
var user_id = "anilthanki"
var repo_name = "phylotree"
var branch = "master"
var dir = "data"
</code>
</pre>
<p><b><i>Note:</i></b> If you have forked the repository without changing the name then just update the <code> user_id </code> to your username in this file.
<h3>
<a id="Output" class="anchor" href="#authors-and-contributors" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Output</h3>
<p> When the pipeline is run for the first time a folder with the taxID name is created that saves all the output files generated by the pipeline. </p>
<p> The main output files generated by the pipeline include:</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th align="center">Contents</th>
</tr>
</thead>
<tbody>
<tr>
<td>taxID.fa</td>
<td align="center">All protein sequences downloaded from NCBI for the specified taxID</td>
</tr>
<tr>
<td>taxID_final_set.fa</td>
<td align="center">The final set of sequences used for Multiple Sequence Alignment</td>
</tr>

<tr>
<td>taxID_tree.nhx</td>
<td align="center">The final tree generated and rerooted by RAxML</td>
</tr>
<tr>
<td>taxID_clustalo_dist_mat.csv</td>
<td align="center">The pairwise distance matrix file </td>
</tr>
<tr>
<td>taxID_metadata</td>
<td align="center">The metadata file for each sequence from taxID_final_set.fa that exists in NCBI </td>
</tr>
</tbody>
</table>

<h3>
<a id="visualisation" class="anchor" href="#authors-and-contributors" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Visualisation</h3>
<p><a href="https://github.com/anilthanki/phylotree"> Phylotree </a> is a visualisation plugin for ViCTree. This is a customised phylogenetic tree visualisation plugin developed using D3.js. It reads input files from data directory of this repository and displays the phylogenetic tree. </p>
<p> Phylotree has a scroll bar at the top that corresponds to the pairwise distances between the nodes of the tree. User can select any value between 0 and 100 based on which clusters within the tree can be highlighted. </p>
<h3>
<a href="example.html"> Densovirinae analysis</a>
</h3>
<p>The ViCTree framework is developed by : </p>
<p>Joseph Hughes (<a href="https://github.com/josephhughes" class="user-mention">@josephhughes</a>), Sejal Modha (<a href="https://github.com/sejmodha" class="user-mention">@sejmodha</a>) and Anil Thanki (<a href="https://github.com/anilthanki" class="user-mention">@anilthanki</a>) </p>

  </body>
</html>
